build.go

```diff
@@ -0,0 +1,355 @@
+package composer
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/paketo-buildpacks/packit/v2"
+	"github.com/paketo-buildpacks/packit/v2/draft"
+	"github.com/paketo-buildpacks/packit/v2/fs"
+	"github.com/paketo-buildpacks/packit/v2/pexec"
+	"github.com/paketo-buildpacks/packit/v2/scribe"
+)
+
+// DetermineComposerInstallOptions defines the interface to get options for `composer install`
+//go:generate faux --interface DetermineComposerInstallOptions --output fakes/determine_composer_install_options.go
+type DetermineComposerInstallOptions interface {
+	Determine() []string
+}
+
+// Executable just provides a fake for pexec.Executable for testing
+//go:generate faux --interface Executable --output fakes/executable.go
+type Executable interface {
+	Execute(pexec.Execution) (err error)
+}
+
+// Calculator defines the interface for calculating a checksum of the given set
+// of file paths.
+//go:generate faux --interface Calculator --output fakes/calculator.go
+type Calculator interface {
+	Sum(paths ...string) (string, error)
+}
+
+func Build(
+	logger scribe.Emitter,
+	composerInstallOptions DetermineComposerInstallOptions,
+	composerInstallExec Executable,
+	composerGlobalExec Executable,
+	checkPlatformReqsExec Executable,
+	path string,
```

Comment:
I'm confused why this needs to be passed an argument to build

Reply:
That was purely for testing purposes. Composer has quite a few environment variables that can be set (https://getcomposer.org/doc/03-cli.md#environment-variables) and so I explicitly wanted user-set environment variables to be passed into the `composer` commands executed here, except for those env vars which explicitly need to be given a value for the buildpack to work.